syntax.cf

```diff
@@ -5,15 +5,21 @@ token SDouble '-'? digit+ '.' digit+ ('e' '-'? digit+)? ;
 token SInt '-' digit+ ;
 token Int digit+ ;
 
--- The regular expression for an element of a tensor variable
-token TensorElement letter (letter|digit|'@')* '_' digit+ ('-' digit+)* ;
-
 -- The regular expression for a tensor variable
-token VariableName letter (letter|digit|'@')* ;
+token VariableName letter (letter|digit|'_')* ;
+
+-- Sequence of integers to represent Tensor shape/indicies
+separator nonempty Int "," ;
+
+-- Production of Tensor Shape (empty shape is allowed for scalar tensors)
+ScalarTensShape. TensorShape ::= ;
```

Comment:
I think better just to allow the user to write `[]` for the tensor shape rather than nothing at all. Much clearer intent!